staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -655,6 +677,44 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 	return s.versioner.UpdateList(listObj, uint64(returnedRV), "", nil)
 }
 
+func splitKvsIntoChunks(concurrencyLevel int, kvs []*mvccpb.KeyValue) [][]*mvccpb.KeyValue {
+	var (
+		chunks    [][]*mvccpb.KeyValue
+		chunkSize = (len(kvs) + concurrencyLevel - 1) / concurrencyLevel
+	)
+
+	for i := 0; i < len(kvs); i += chunkSize {
+		end := i + chunkSize
+		if end > len(kvs) {
+			end = len(kvs)
+		}
+		chunks = append(chunks, kvs[i:end])
+	}
+
+	return chunks
+}
+
+func (s *store) transformChunk(ctx context.Context, kvPairs []*mvccpb.KeyValue) <-chan *transformedObj {
```

Comment:
There's a slight problem with this approach: if `kvPairs[0]` transformation takes a long time, all `kvPairs[1:]` will be delayed too.
You could instead feed `mvccpb.KeyValue` items to these workers over a single shared channel. Then, when one transformation is slow it doesn't delay the rest.

Reply:
This is how I implemented this initially (one shared channel), but changed to this somewhat slower option to preserve the order - I believe this is a requirement.

Reply:
You could pass the index along with `transformedObj`, but either way is fine.
Please leave a comment about requirement to preserve the order though.

Reply:
Added a comment.